jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/status/CodeWhispererStatusBarWidget.kt

```diff
@@ -46,12 +48,16 @@ class CodeWhispererStatusBarWidget(project: Project) :
 
     override fun getSelectedValue(): String = message("codewhisperer.statusbar.display_name")
 
-    override fun getIcon(): Icon =
-        if (CodeWhispererInvocationStatus.getInstance().hasExistingInvocation()) {
+    override fun getIcon(): Icon {
+        val connectionType = CodeWhispererExplorerActionManager.getInstance().checkActiveCodeWhispererConnectionType(project)
+        return if (connectionType == CodeWhispererLoginType.Expired) {
+            AllIcons.General.BalloonWarning
+        } else if (CodeWhispererInvocationStatus.getInstance().hasExistingInvocation()) {
```

Comment:
Do we also want to add Bearer connection listener to listen for the connection change (to/from expired state) to sync the icon with the connection status?

Reply:
good catch, will add